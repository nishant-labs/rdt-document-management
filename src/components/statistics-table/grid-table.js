import React from 'react';
import { AgGridColumn, AgGridReact } from 'ag-grid-react';
import { TypographyWithBox } from '../TypographyWithBox';
import { selectTableData } from '../../context/shared/selectors';
import { currencyFormatter, numberFormatter } from '../../utils/number';

import 'ag-grid-community/dist/styles/ag-grid.css';
import 'ag-grid-community/dist/styles/ag-theme-alpine.css';

import './table-styles.scss';

const defaultColDef = {
  sortable: true,
  filter: true,
};

const gdpCurrencyFormatter = (cellData) => {
  if (cellData.value) {
    return currencyFormatter(cellData.value);
  }
  return 'Not available';
};

const populationFormatter = (cellData) => {
  if (cellData.value) {
    return numberFormatter(cellData.value);
  }
  return 'Not available';
};

export const GridTableContainer = ({ population, gdp }) => {
  const rowData = selectTableData({ population, gdp });
  if (rowData.length === 0) {
    return (
      <TypographyWithBox type="h6" text="Data not found at World Bank center" />
    );
  }
  return (
    <div className="ag-theme-alpine" style={{ height: 400, width: 900 }}>
      <AgGridReact rowData={rowData} defaultColDef={defaultColDef}>
        <AgGridColumn
          headerName="Country"
          field="countryName"
          width={400}
          sort="asc"
        ></AgGridColumn>
        <AgGridColumn
          headerName="Population"
          field="population"
          width={250}
          valueFormatter={populationFormatter}
        ></AgGridColumn>
        <AgGridColumn
          headerName="GDP (US dollars)"
          field="gdp"
          width={250}
          valueFormatter={gdpCurrencyFormatter}
        ></AgGridColumn>
      </AgGridReact>
    </div>
  );
};
