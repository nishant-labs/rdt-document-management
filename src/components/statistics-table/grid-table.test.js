import React from 'react';
import { shallow } from 'enzyme';
import { GridTableContainer } from './grid-table';

const setupTest = (setupProps = {}) => {
	const wrapper = shallow(<GridTableContainer />);
	return {
		wrapper,
	};
};

describe('components/statistics-table/grid-table', () => {
	test('should render', () => {
		const { wrapper } = setupTest();
		expect(wrapper.debug()).toMatchSnapshot();
	});
});
