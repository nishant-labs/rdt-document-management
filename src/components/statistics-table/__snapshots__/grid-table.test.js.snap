// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/statistics-table/grid-table should render 1`] = `"<TypographyWithBox type=\\"h6\\" text=\\"Data not found at World Bank center\\" />"`;
